tests/unit/vdom.ts

```diff
@@ -1421,6 +1427,28 @@ describe('vdom', () => {
 			assert.strictEqual(root.children[0].bar, 'bar');
 			assert.isFalse(appendChildSpy.called);
 		});
+
+		it('will append nodes with attributes already attached', (test) => {
+			const expected = '<div data-attr="test"></div>';
+			const appendedHtml: string[] = [];
+
+			const createElement = document.createElement.bind(document);
+			const createElementStub = stub(document, 'createElement', (name: string) => {
+				const node = createElement(name);
+				const appendChild = node.appendChild.bind(node);
+				stub(node, 'appendChild', (node: Element) => {
+					appendedHtml.push(node.outerHTML);
+					return appendChild(node);
+				});
+				return node;
+			});
+			spys.push(createElementStub);
+			const projection = dom.create(v('div', { 'data-attr': 'test' }), projectorStub);
+
+			assert.strictEqual(projection.domNode.innerHTML, expected);
+			assert.lengthOf(appendedHtml, 1);
```

Comment:
given this PR is about an ordering change, it would've been nice to have a slightly more expressive assertion for that. ie sinon keeps track of call orders across spies/stubs so we could've done something like:
```ts
assert.isTrue(appendChildSpy.calledAfter(setAttributeSpy));
```
anyway this isn't blocking us taking this 👍.